catkin_ws/src/pdc_ros/src/pdc_ros/pdc_ros_simple_streaming.py

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env python
+import os
+import numpy as np
+
+# ROS imports
+import rospy
+
+from utils import *
+
+# pdc_ros_msgs
+import pdc_ros_msgs.msg
+
+import time
+import sys
+import cv2
+from std_msgs.msg import String
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge, CvBridgeError
+from PIL import Image as PILImage
+
+import dense_correspondence_manipulation.utils.utils as pdc_utils
+pdc_utils.add_dense_correspondence_to_python_path()
+dc_source_dir = pdc_utils.getDenseCorrespondenceSourceDir()
+from dense_correspondence.evaluation.evaluation import DenseCorrespondenceEvaluation
+from dense_correspondence.evaluation.plotting import normalize_descriptor
+
+import torch
+
+
+NETWORK_CONFIG_FILENAME = os.path.join(get_config_directory(), 'trained_networks.yaml')
+NETWORK_NAME = "shoes_consistent_M_background_0.500_3"
+RGB_TOPIC = "/camera_carmine_1/rgb/image_rect_color"
+
+class StreamingPdcRos(object):
+
+    def __init__(self):
+        self.bridge = None
+        self.load_dcn_network()
+
+    def load_dcn_network(self):
+        """
+        Loads the DCN.
+        """
+        config = pdc_utils.getDictFromYamlFilename(NETWORK_CONFIG_FILENAME)
+        defaults_config = pdc_utils.get_defaults_config()
+        pdc_utils.set_cuda_visible_devices([0])
+        dce = DenseCorrespondenceEvaluation(config)
+
+        self.dcn = dce.load_network_from_config(NETWORK_NAME)
```

Comment:
you might want to set the network to `test` mode. I think by default it is in `train` mode.

Reply:
added new commit which puts into eval mode